Migrates features-tab to release coordinator page and introduces a new tab for platform-params on admin page

lkbhitesh07 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of N/A.
2. This PR does the following: As part of the [project: Normalize the usage of platform-parameters](https://docs.google.com/document/d/1ndjDml4PUB_LA8mwwTQOawDFOWyNnhOiPXpAWaK2cXI/edit) this PR performs 3 things:
(a) Removes some of the config properties and updates the codebase accordingly
(b) Moves the `features-tab` on the admin page to the release-coordinator page and adds a `platform-parameter-tab` on the admin page.
(c) Creates platform-parameters / feature-flags of the config properties(Please note that the config properties are still in use, the next PR will migrate the config properties to params/features.)

The list of config properties that are being removed are:
- classroom_promos_are_enabled
- contributor_dashboard_is_enabled
- max_number_of_explorations_in_math_svgs_batch
- max_number_of_svgs_in_math_svgs_batch

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
**We no longer have the config properties we removed from the config tab and the rest config properties work fine.**
https://github.com/oppia/oppia/assets/44300735/69d1e671-790b-4d82-b9eb-409601873150


**The features-tab has been migrated to the release-coordinator page and it works fine.**
https://github.com/oppia/oppia/assets/44300735/5a08bdb5-e219-486b-a64b-55f09a0eff7d


**The new tab for platform-parameter has been added to the admin page and the functionality works fine.**
https://github.com/oppia/oppia/assets/44300735/87cc4dc9-b781-4cd9-b1c9-1173c16f52e6


## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

lkbhitesh07 committed:
added another data type to the platform parameter

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into new-type-plat-param

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into new-type-plat-param

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into new-type-plat-param

lkbhitesh07 committed:
added one off jobs

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into new-type-plat-param

lkbhitesh07 committed:
All jobs works perfectly and have been tested on the local server

lkbhitesh07 committed:
fixing lint issues

lkbhitesh07 committed:
removed the jobs

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
Merge branch 'new-type-plat-param' of github.com:lkbhitesh07/oppia into migrate-config-props

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
migrating config properties

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
removed classroom-promos config property

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
removed checkpoints-cofig-property

lkbhitesh07 committed:
removed contributor-dashboard config property

lkbhitesh07 committed:
removed ask-learners-for-answers config property

lkbhitesh07 committed:
removed max-number-of-exps and max-number-of-math-svgs config properties

lkbhitesh07 committed:
fixed backend test

lkbhitesh07 committed:
fixed frontend tests

lkbhitesh07 committed:
fixed acceptance test

lkbhitesh07 committed:
fixed some e2e tests

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
fixed python type checks

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
added the frontend code for platform params

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
added platform params

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
added code to platform ts file

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
removed unneeded changes

lkbhitesh07 committed:
nit fixes

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
platform-param page working correctly

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
fixed backend tests

lkbhitesh07 committed:
indentation and name correction

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
fixed a few tests

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
fixed a few lints

lkbhitesh07 committed:
fixed backend tests

lkbhitesh07 committed:
added and fixed backend tests

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
fixed e2e and some frontend tests

lkbhitesh07 committed:
added frontend tests for platform params

lkbhitesh07 committed:
Merge branch 'develop' of github.com:oppia/oppia into migrate-config-props

lkbhitesh07 committed:
restored ask-learners-for-answers config property

lkbhitesh07 committed:
nits

lkbhitesh07 committed:
fixed lints

lkbhitesh07 committed:
fixed more lints

lkbhitesh07 committed:
a bit more lint

lkbhitesh07 committed:
nit

lkbhitesh07 committed:
resolved backend test

lkbhitesh07 committed:
fixed frontend tests

lkbhitesh07 committed:
restored checkpoint feature config property

lkbhitesh07 committed:
fixed lint

lkbhitesh07 committed:
fixed frontend tests and some more nits

lkbhitesh07 committed:
fixed lint

lkbhitesh07 committed:
fixed frontend tests and did some nits

lkbhitesh07 committed:
fixed lint

lkbhitesh07 commented:
Hey @seanlip, can you please take a first pass?

Few points:
- Previously I had migrated more config properties than the current but that were resulting in lots of test failures which was making it hard to debug so I will migrate the config properties in small chunks in the upcoming PRs which will make it easy for the debugging.
- Please note that some of the e2e tests will fail because of the migration of the features page from admin to release-coordinator. I'm working on it and will do that by today. Please note that I was not able to find the e2e tests for the release-coordinator page and as I have shifted the `features-tab` to the release-coordinator page bunch of e2e tests are failing. Does this mean I have to introduce a new e2e test file for this?

I have added the proof of work so that we can confirm the functionality works fine and that is why I have requested for the first pass.
Thanks

lkbhitesh07 committed:
fixed the frontend coverage coverage for platform-param component file

seanlip commented:
Hi @lkbhitesh07 -- yup, I think you will need to refactor the e2e tests a bit to bring the functionality that was previously in the features tab page to a new file created for the release coordinator page instead. I think the reason we don't have an existing e2e file for that is because the stuff on the RC page involves running jobs which we can't really test in e2e due to the time lag -- but now that we also use the RC page for setting feature flags, we will need a webdriverio utils file for that page, similar to what we have for the other pages. Does this answer your question?

Also, I looked at the videos. Great job! Some questions:
- I see some properties in both the config and platform params tab (in the first and third videos). Shouldn't they be in one or the other?
- For the third video, I suggest having a demonstration that setting the platform parameter actually works and has the same effect as setting the corresponding config property -- i.e., use something like classroom_promos that's not "dummy".  (In fact you can probably remove the dummy entirely.) Show that turning it off results in one effect for end users, and turning it on results in the opposite effect.

lkbhitesh07 commented:
Thanks @seanlip, that does answer my question. I will try to get it done asap.

Thanks for the review
> I see some properties in both the config and platform params tab (in the first and third videos). Shouldn't they be in one or the other?

Yes, this is intentional. Well, this PR mainly aims to have a page for platform parameters and we try to verify that we are able to populate the page correctly. At this point, the config properties are still in use and the codebase still uses the config properties instead of platform parameters. The upcoming PRs will be migrating the config properties to platform parameters in chunks and there we will be removing the properties from the config tab and we will only have on parameters tab.

> For the third video, I suggest having a demonstration that setting the platform parameter actually works and has the same effect as setting the corresponding config property -- i.e., use something like classroom_promos that's not "dummy". (In fact you can probably remove the dummy entirely.) Show that turning it off results in one effect for end users, and turning it on results in the opposite effect.

Yeah, I think I can do this part for a single property to verify that the changes work correctly. I initially was verifying that the changes works correctly purely on the basis of the data getting created in the datastore.

Thanks

seanlip commented:
> Yes, this is intentional. Well, this PR mainly aims to have a page for platform parameters and we try to verify that we are able to populate the page correctly. At this point, the config properties are still in use and the codebase still uses the config properties instead of platform parameters. The upcoming PRs will be migrating the config properties to platform parameters in chunks and there we will be removing the properties from the config tab and we will only have on parameters tab.

But I don't understand ... what are we actually doing in this PR? I thought this PR migrates 4 properties, so I would expect to see 4 properties on the parameters tab in the demo and those 4 properties removed from the config tab.

lkbhitesh07 commented:
> But I don't understand ... what are we actually doing in this PR? I thought this PR migrates 4 properties, so I would expect to see 4 properties on the parameters tab in the demo and those 4 properties removed from the config tab.

Oh, let me explain it a bit. Those 4 properties are the ones that need to be removed completely from the codebase and update the codebase with the appropriate values. For example for `classroom_promos_are_enabled`, we want to turn this part `on` completely, so I have removed the config property and updated the codebase to always show classroom promos.
You can also have a look at the `Technical hints` section on the project description, there we have this part under `The following config properties should be removed:`. [Project](https://docs.google.com/document/d/1ndjDml4PUB_LA8mwwTQOawDFOWyNnhOiPXpAWaK2cXI/edit)
Initially, I was performing this part altogether which is to remove all the unwanted config properties but they were resulting in lots of test failures and that is why I plan to do that in the upcoming PR. Let me know in case you have more questions.

Thanks

EDIT: We mainly perform 3 things
(a) Removes some of the config properties and updates the codebase accordingly(explained above)
(b) Moves the features-tab on the admin page to the release-coordinator page and adds a platform-parameter-tab on the admin page.
(c) Creates platform-parameters / feature-flags of the config properties(Please note that the config properties are still in use, the next PR will migrate the config properties to params/features, basically the backend and frontend still use config properties.)

seanlip commented:
Ah, ok. Could you separate (a) into one PR and (b)+(c) into another? That will make it easier ... the current PR has 2.5k+ lines and is difficult to digest.

For (b)+(c) I am a bit confused. I don't think you should push a PR to develop that has the same item existing as both config property and feature flag, and it also seems weird to have an empty feature flag tab. Why not do it step by step, like this:

1. Move feature flags to /release-coordinator, get rid of features tab completely from /admin
2. Set up platform parameters tab in /admin and migrate a single config property to it to verify that it works
3. Migrate the rest of the config properties, either all at once or in batches

lkbhitesh07 commented:
Closing this PR as we decided to split the PR into 4 separate PRs.
Thanks
